FirebaseAuthUI/AccountManagement/FUIAccountSettingsViewController.m

```diff
@@ -0,0 +1,421 @@
+//
+//  Copyright (c) 2017 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "FUIAccountSettingsViewController.h"
+
+#import <FirebaseAuth/FirebaseAuth.h>
+
+#import "FUIAuthStrings.h"
+#import "FUIStaticContentTableViewController.h"
+#import "FUIAccountSettingsOperation.h"
+#import "FUIAccountSettingsOperationDeleteAccount.h"
+#import "FUIAccountSettingsOperationForgotPassword.h"
+#import "FUIAccountSettingsOperationSignOut.h"
+#import "FUIAccountSettingsOperationUnlinkAccount.h"
+#import "FUIAccountSettingsOperationUpdateEmail.h"
+#import "FUIAccountSettingsOperationUpdateName.h"
+#import "FUIAccountSettingsOperationUpdatePassword.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** @var FUIASAccountState
+    @brief Defines all possible states of current loogged-in @c FIRUser.
+ */
+typedef NS_ENUM(NSInteger, FUIASAccountState) {
+  FUIASAccountStateUnknown = 0,
+  FUIASAccountStateEmailPassword,
+  FUIASAccountStateLinkedAccountWithEmail,
+  FUIASAccountStateLinkedAccountWithoutEmail,
+  FUIASAccountStateLinkedAccountWithEmailPassword
+};
+
+/** @var kUserAccountImage
+    @brief Name of icon to show default user account.
+ */
+static NSString *const kUserAccountImage = @"ic_account_circle.png";
+
+@interface FUIAccountSettingsViewController () <FUIAccountSettingsOperationUIDelegate>
+@end
+
+@implementation FUIAccountSettingsViewController {
+  __unsafe_unretained IBOutlet UITableView *_tableView;
```

Comment:
Adding to this, if the IBOutlet is owned entirely by the view controller (which it should be), you can go ahead and make this a strong reference.